src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py

```diff
@@ -0,0 +1,401 @@
+    }
+
+
+def _create_postgresql_connection_strings(host, user, password, database):
+    result = {
+        'psql_cmd': "psql --host={host} --port=5432 --username={user} --dbname={database}",
+        'ado.net': "Server={host};Database={database};Port=5432;User Id={user};Password={password};",
+        'jdbc': "jdbc:postgresql://{host}:5432/{database}?user={user}&password={password}",
+        'jdbc Spring': "spring.datasource.url=jdbc:postgresql://{host}:5432/{database}  "
+                       "spring.datasource.username={user}  "
+                       "spring.datasource.password={password}",
+        'node.js': "var client = new pg.Client('postgres://{user}:{password}@{host}:5432/{database}');",
+        'php': "host={host} port=5432 dbname={database} user={user} password={password}",
+        'python': "cnx = psycopg2.connect(database='{database}', user='{user}', host='{host}', password='{password}', "
+                  "port='5432')",
+        'ruby': "cnx = PG::Connection.new(:host => '{host}', :user => '{user}', :dbname => '{database}', "
+                ":port => '5432', :password => '{password}')",
+    }
+
+    connection_kwargs = {
+        'host': host,
+        'user': user,
+        'password': password if password is not None else '{password}',
+        'database': database
+    }
+
+    for k, v in result.items():
+        result[k] = v.format(**connection_kwargs)
+    return result
+
+
+def _create_postgresql_connection_string(host, password):
+    connection_kwargs = {
+        'host': host,
+        'password': password if password is not None else '{password}'
+    }
+    return 'postgres://postgres:{password}@{host}/postgres?sslmode=require'.format(**connection_kwargs)
+
+
+def _form_response(username, sku, location, id, host, version, password, connection_string, firewall_id=None, subnet_id=None):
+    '''
+    from collections import OrderedDict
+    new_entry = OrderedDict()
+    new_entry['Id'] = id
+    if subnet_id is not None:
+        new_entry['SubnetId'] = subnet_id
+    new_entry['Location'] = location
+    new_entry['SkuName'] = sku
+    new_entry['Version'] = version
+    new_entry['Host'] = host
+    new_entry['UserName'] = username
+    new_entry['Password'] = password
+    new_entry['ConnectionString'] = connection_string
+    if firewall_id is not None:
+        new_entry['FirewallName'] = firewall_id
+    '''
+    output = {
+        'host': host,
+        'username': username,
+        'password': password,
+        'skuname': sku,
+        'location': location,
+        'id': id,
+        'version': version,
+        'connectionString': connection_string
+    }
+    if firewall_id is not None:
+        output['firewallName'] = firewall_id
+    if subnet_id is not None:
+        output['subnetId'] = subnet_id
+    return output
+
+
+def _update_local_contexts(cmd, server_name, resource_group_name, location, user):
```

Comment:
Same with previous comments.

Reply:
Same as comment to previous comment: We automatically generate some arguments without users' input (e.g., server name, username etc). In this case, we weren't able to save the local context values just by defining the attribute. Is there another way we can set it?